src/proxy/http/metrics/report.rs

```diff
@@ -120,6 +119,28 @@ where
         Ok(())
     }
 
+    fn fmt_by_status<M, F>(
+        &self,
+        f: &mut fmt::Formatter,
+        metric: Metric<M>,
+        get_metric: F,
+    ) -> fmt::Result
+    where
+        M: FmtMetric,
+        F: Fn(&StatusMetrics<C>) -> &M,
+    {
+        for (tgt, tm) in &self.by_target {
+            if let Ok(tm) = tm.lock() {
+                for (status, m) in &tm.by_status {
```

Comment:
TIOLI: if you bind this like
```suggestion
                for (&status, ref m) in &tm.by_status {
```
then you don't need to dereference later:
```rust
                    let labels = (tgt, Status(status));
                    get_metric(m).fmt_metric_labeled(f, metric.name, labels)?;
```